sim-lib/src/lib.rs

```diff
@@ -275,13 +277,48 @@ impl Simulation {
         Ok(())
     }
 
+    // validate_node_network validates that the nodes that are on the same bitcoin network.
+    // e.g. node A and B are on testnet, node C is on mainnet should return an error.
+    async fn validate_node_network(&self) -> Result<(), LightningError> {
+        let mut valid_network = Option::None;
+        let mainnet = Network::Bitcoin.to_string().to_lowercase();
+
+        for node in self.nodes.values() {
+            let network = node.lock().await.get_info().network.clone();
+            if network == mainnet {
+                return Err(LightningError::ValidationError(
+                    "mainnet is not supported".to_string(),
+                ));
+            }
+
+            if let Some(valid) = &valid_network {
+                if valid != &network {
+                    return Err(LightningError::ValidationError(format!(
+                        "nodes are not on the same network {}",
+                        network,
+                    )));
+                }
+            } else {
+                valid_network = Some(network);
+            }
```

Comment:
We could shorten this a little with `valid_network = valid_network.unwrap_or(network.clone());` on 294, lets us get rid of the conditional / fill our own value if it's currently none.